src/elfpy/pricing_models/base.py

```diff
@@ -382,72 +382,16 @@ def get_max_long(
             If they aren't document why they aren't.
             Otherwise, remove them.
         """
-        available_bonds = market_state.bond_reserves - market_state.bond_buffer
-        if available_bonds <= 0:
-            return 0, 0
-        last_maybe_max_long = 0, 0
-        bond_percent = 1
-        num_iters = 25
-        for step_size in [1 / (2 ** (x + 1)) for x in range(num_iters)]:
-            # Compute the amount of base needed to purchase the specified amount
-            # of bonds.
-            trade_result = self.calc_in_given_out(
-                out=Quantity(amount=available_bonds * bond_percent, unit=TokenType.PT),
+        epsilon = 1e-7
+        k = pow(market_state.share_reserves,1-time_remaining.stretched_time) + pow(2*market_state.bond_reserves+market_state.share_reserves,1-time_remaining.stretched_time)
```

Comment:
don't we need initial_share_price and share_price on the left hand side of this?
`k = (share_price * initial_share_price) * pow(initial_share_price * market_state.share_reserves, 1 - time_remaining.stretched_time)...`

Reply:
ya